src/components/IconList/IconList.js

```diff
@@ -0,0 +1,174 @@
+import React, { PureComponent, createRef } from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import List from '@geops/react-ui/components/List';
+import Button from '@geops/react-ui/components/Button';
+import { FaCaretUp, FaCaretDown } from 'react-icons/fa';
+import { withTranslation } from 'react-i18next';
+import { compose } from 'lodash/fp';
+
+import './IconList.scss';
+
+const propTypes = {
+  disabled: PropTypes.bool,
+  options: PropTypes.array.isRequired,
+  selected: PropTypes.string,
+  icons: PropTypes.object.isRequired,
+
+  /**
+   * Icon size in pixels
+   */
+  iconSize: PropTypes.shape({
+    height: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    width: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+  }),
+
+  onSelect: PropTypes.func.isRequired,
+  displayOnTop: PropTypes.bool,
+  t: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+  disabled: false,
+  selected: null,
+  iconSize: {
+    height: 16,
+    width: 42,
+  },
+  displayOnTop: false,
+};
+
+class IconList extends PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.ref = createRef();
+
+    this.state = {
+      selectedOption: props.selected,
+      iconListVis: false,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { selected } = this.props;
+
+    if (selected !== prevProps.selected) {
+      this.select(selected);
+    }
+  }
+
+  select(option) {
+    this.setState({
+      selectedOption: option,
+    });
+  }
+
+  renderOption(option) {
+    const { icons, t, options, iconSize } = this.props;
+
+    const opt = option || (options.length ? options[0] : null);
+
+    const label = <div className="wkp-icon-list-label">{t(opt)}</div>;
+
+    return opt && icons[opt] ? (
+      <>
+        <div className="wkp-icon-list-icon">
+          <img
+            src={icons[option]}
+            height={iconSize.height}
+            width={iconSize.width}
+            draggable="false"
+            alt={t('Kein Bildtext')}
+          />
+        </div>
+        {label}
+      </>
+    ) : (
+      label
+    );
+  }
+
+  renderList() {
+    const { displayOnTop, options, onSelect } = this.props;
+
+    const list = (
+      <List
+        items={options}
+        className="tm-list wkp-icon-list-list"
+        renderItem={option => this.renderOption(option)}
+        getItemKey={option => option}
+        onSelect={(e, option) => {
+          onSelect(option);
+          this.setState({
+            iconListVis: false,
+            selectedOption: option,
+          });
+        }}
+      />
+    );
+
+    if (displayOnTop) {
+      const rect = this.ref.current.getBoundingClientRect();
+      const listStyles = {
+        top: `${rect.y + rect.height}px`,
+        left: `${rect.x}px`,
+      };
+
+      const targetElements = document.getElementsByClassName(
+        'tm-trafimage-maps',
```

Comment:
@oterral I'm not sure if this a good idea to reference this element. I would like an element which is in our scope but similar to body. So I can add this dropdown list to the very top of all elements.

However, this class name is also used in other components (css).

Reply:
@gevos does it matter if it's in our scope ?  document.body is fine I guess. Because it will be easier to get/set a position form/to the body. if you need to have the position depending on trafimage-maps elt , i will be more complicated. Try with document.body for now. We will after if we need more.

Reply:
@oterral What about the WebComponent? I thought, it should be possible to integrate a trafimage-maps application into another page without using an IFrame. Adding an element to the body could have side-effects.

Reply:
@oterral And there is another thing: I cannot use the styles from IconList.scss as they are in a scope of the class s466903671.

Reply:
Yes,  because of the styles is not possible I guess.